src/components/instructor/AssignmentUpdate.js

```diff
@@ -1,17 +1,88 @@
 import React, { useState } from 'react';
+import {SERVER_URL} from "../../Constants";
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import TextField from "@mui/material/TextField";
+import DialogActions from "@mui/material/DialogActions";
 
 //  instructor updates assignment title, dueDate 
 //  use an mui Dialog
 //  issue PUT to URL  /assignments with updated assignment
 
 const AssignmentUpdate = (props)  => {
 
+  const [open, setOpen] = useState(false);
+  const [editMessage, setEditMessage] = useState('');
+  const [assignment, setAssignment] = useState(
+      {id:'', title:'', dueDate:'', courseId:'', secId:'', secNo:''});
 
-    return (
-        <>
-          <h3>Not implemented</h3>  
-        </>                       
-    )
+  /*
+   *  dialog for edit assignment
+   */
+  const editOpen = () => {
+    setEditMessage('');
+    setAssignment({id:'', title:'', dueDate:'', courseId:'', secId:'', secNo:''});
+    setOpen(true);
+    setAssignment(props.assignment);
+  };
+
+  const editClose = () => {
+    setOpen(false);
+    props.onClose();
+  };
+
+  const editChange = (event) => {
+    setAssignment({...assignment,  [event.target.name]:event.target.value})
+  }
+
+  const onSave = () => {
+    saveAssignment(assignment);
```

Comment:
An empty title shouldn't be allowed, though this should probably be handled on the back end.

Reply:
Updated on previous assignment-2, backend.